### DISCUSSION TITLE
Non-anonymized local repo / anonymized remote one - Possible?

### DISCUSSION THREAD

**Discussion by Lapis-Lazuli** [community/community#61820] [New to GitHub]
Hi,

**_Context and objective:_**

We are investing using GitHub to share a customer's projects information among customer's projects members. For increased security, GitHub-shared info should be anonymized before being uploaded. to the remote repo.

The anonymization / de-anonymization process should be as transparent as possible for the project's members, who would work locally on non-anonymized repos.

We can develop and manage locally the anonymization (before commit) and de-anonymization (after pulling) code . 

**_Investigations_**

To do it, we looked at various options, such as: 

- Use GitHub actions. But our understanding it that they run on the remote repo, so the anonymization code wouldn't remain local. Correct?
- Use nektos/act? But it seems to rely on GitHub Workflows defined remotely. Would that work with anonymization code locally? And how practical would that be?

**_Question_**
What else would you recommend?

Thank you.

**Comment by GhostOf0days** [comment]
Why can't you use private repos?

**Comment by Lapis-Lazuli** [comment]
> Why can't you use private repos?

The remote repo is already private. We're looking for extra-security.

**Comment by Lapis-Lazuli** [comment]
Any other answer?

**Reply by GhostOf0days** [reply]
You could look into [Github Enterprise](https://github.com/enterprise).

**Reply by GhostOf0days** [reply]
Specifically, Github Enterprise Server.

**Reply by Lapis-Lazuli** [reply]
@GhostOf0days: thank you, I will give it a look.

**Reply by GhostOf0days** [reply]
Could you also please close this post as soon as you are able to get it working?

**Comment by Lapis-Lazuli** [comment]
> Could you also please close this post as soon as you are able to get it working?

If I undertsand well, Github Enterprise is not free. In this context, the buget - as well as the security policies - are outside of our control, unfortunately. So this can't be used.

Maybe the absence of solutions shows that, in our context, there aren't. In which case, if there are no other answers in a couple of days, I'll close the discussion.

**Reply by GhostOf0days** [reply]
Unfortunately, there are no other solutions on Github. You would have to look for an outside solution or develop an in-house solution from scratch. I suggest looking into Bitbucket or Gitlab instead of Github.

**Comment by Lapis-Lazuli** [comment]
> Unfortunately, there are no other solutions on Github. You would have to look for an outside solution or develop an in-house solution from scratch. I suggest looking into Bitbucket or Gitlab instead of Github.

Thanks. Closing the discussion then.
